nav2_costmap_2d/src/clear_costmap_service.cpp

```diff
@@ -70,6 +85,12 @@ void ClearCostmapService::clearExceptRegion(const double reset_distance)
   }
 }
 
+void ClearCostmapService::clearEntirely()
+{
+  std::unique_lock<Costmap2D::mutex_t> lock(*(costmap_.getCostmap()->getMutex()));
```

Comment:
I think you should take the mutex in resetLayers. IMHO deciding whether to lock or not is a private detail of a class, and shouldn't be imposed on it's clients.

Also, should it have been acquired for the clearExceptRegion call as well, or is that safe for some other reason?

Reply:
Agree on the principle, in this case, I was following the same implementation as in [move_base](https://github.com/ros-planning/navigation/blob/a6f3d4b49335b6de5c33044dd2123df0f491453f/move_base/src/move_base.cpp#L330-L338).

We also need to lock when clearing except a region, in that case, it is acquired when clearing a particular layer in `clearLayerExceptRegion` as done in [move_base](https://github.com/ros-planning/navigation/blob/a6f3d4b49335b6de5c33044dd2123df0f491453f/clear_costmap_recovery/src/clear_costmap_recovery.cpp#L125)